Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift

```diff
@@ -0,0 +1,115 @@
+import Foundation
+import XcodeProj
+
+public protocol DefaultSettingsProviding {
+    func defaultProjectSettings(project: Project,
+                                buildConfiguration: BuildConfiguration) -> [String: Any]
+
+    func defaultTargetSettings(target: Target,
+                               buildConfiguration: BuildConfiguration) -> [String: Any]
+}
+
+public final class DefaultSettingsProvider: DefaultSettingsProviding {
+    private static let essentialProjectSettings: Set<String> = [
+        "ALWAYS_SEARCH_USER_PATHS",
+        "DEBUG_INFORMATION_FORMAT",
+        "ENABLE_NS_ASSERTIONS",
+        "ENABLE_TESTABILITY",
+        "GCC_DYNAMIC_NO_PIC",
+        "GCC_OPTIMIZATION_LEVEL",
+        "GCC_PREPROCESSOR_DEFINITIONS",
+        "MTL_ENABLE_DEBUG_INFO",
+        "ONLY_ACTIVE_ARCH",
+        "CLANG_ANALYZER_NONNULL",
+        "CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION",
+        "CLANG_CXX_LANGUAGE_STANDARD",
+        "CLANG_CXX_LIBRARY",
+        "CLANG_ENABLE_MODULES",
+        "CLANG_ENABLE_OBJC_ARC",
+        "CLANG_ENABLE_OBJC_WEAK",
+        "COPY_PHASE_STRIP",
+        "ENABLE_STRICT_OBJC_MSGSEND",
+        "GCC_C_LANGUAGE_STANDARD",
+        "GCC_NO_COMMON_BLOCKS",
+        "PRODUCT_NAME",
+    ]
+
+    private static let essentialTargetSettings: Set<String> = [
+        "SDKROOT",
+        "CODE_SIGN_IDENTITY",
+        "LD_RUNPATH_SEARCH_PATHS",
+        "VALIDATE_PRODUCT",
+        "SWIFT_OPTIMIZATION_LEVEL",
+        "SWIFT_ACTIVE_COMPILATION_CONDITIONS",
+        "CURRENT_PROJECT_VERSION",
+        "DEFINES_MODULE",
+        "DYLIB_COMPATIBILITY_VERSION",
+        "DYLIB_CURRENT_VERSION",
+        "DYLIB_INSTALL_NAME_BASE",
+        "INSTALL_PATH",
+        "PRODUCT_NAME",
+        "SKIP_INSTALL",
+        "VERSION_INFO_PREFIX",
+        "VERSIONING_SYSTEM",
+        "TARGETED_DEVICE_FAMILY",
+        "EXECUTABLE_PREFIX",
+        "COMBINE_HIDPI_IMAGES",
+        "ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES",
+        "WRAPPER_EXTENSION",
+    ]
+
+    private let settingsHelper = SettingsHelper()
+
+    public init() {}
+
+    // MARK: - DefaultSettingsProviding
+
+    public func defaultProjectSettings(project: Project,
+                                       buildConfiguration: BuildConfiguration) -> [String: Any] {
```

Comment:
Maybe you can do:

```swift
public func defaultProjectSettings(project: Project,
                                       buildConfiguration: BuildConfiguration,
                                       settingsHelper: SettingsHelper = SettingsHelper()) -> [String: Any] {
  // Your logic
}
```

Reply:
It's probably better to not leak `SettingsHelper` type to the `DefaultSettingsProviding` protocol. Some implementations might not needed it at all.